### DISCUSSION TITLE
Add an "existing translations" functionality to globals and collections

### DISCUSSION THREAD

**Discussion by yandodov** [payloadcms/payload#5617] [Feature Requests & Ideas]
It'd be useful to be able to know the existing translations of a Payload document, e.g. when generating [hreflang tags](https://ahrefs.com/blog/hreflang-tags/). This could be done with a predicate function that the user provides, which determines whether a locale exists for a certain document or not.

Let's say I have a global and if that global's `content` field is not empty for a certain locale, I want it to count as having a translation in that locale. I could do it like so:

```js
export const MyGlobal = {
  slug: "my-global",
  translationPredicate: (doc, locale) => !!doc.content[locale.code],
  fields: [
    {
      name: "content",
      type: "textarea",
      localized: true,
    },
  ],
};
```

Then, if my config looks like this:

```js
import { buildConfig } from "payload/config";

export default buildConfig({
  localization: {
    locales: ["en", "es", "fr", "de", "tr", "bg"],
    defaultLocale: "en",
    fallback: true,
  },
});
```

…but my global's data looks like this:

```json
{
  "content": {
    "en": "hello",
    "es": "hola",
    "bg": "здравей"
  }
}
```

…I'll be able to get the relevant translations like so:

```js
const global = await payload.findGlobal({
  slug: "my-global",
});
console.log(global.existingTranslations); // ["en", "es", "bg"]
```

This way I can avoid processing the `fr` and `de` locales, as the document doesn't have translations in those.

**Comment by tim-the-arcane** [comment]
Yeah, this would be really helpful. The current philosophy with field based localization really only makes sense if every single doc and global has their translations from the start. But in reality this rarely is the case. 

I would even go a step further and consider a document only as existing translation if every field with the `required` attribute has content.

This would also help clean up the collection's list view in the admin panel with all the empty title fields.

At the moment one would have to manually create these filtering rules for every collection and also keep those up-to-date when fields change.

**Reply by yandodov** [reply]
Maybe there should be a `localizedPredicate` option which is a user-defined function that receives the current state of the document and returns `true` or `false` to signify whether the document should be deemed translated in a certain locale or not. Payload then saves the result in the database and queries against it. Or the user queries against it.

**Comment by akantic** [comment]
+1, would love to have this, as I currently have to do manually

To be honest, localization sometimes just does not make sense as it can return an empty document for some locale just because that locale was configured, even though it's missing all the fields.
